Sources/Fuzzilli/Fuzzer.swift

```diff
@@ -486,21 +488,37 @@ public class Fuzzer {
 
     /// Process a program that has interesting aspects.
     func processInteresting(_ program: Program, havingAspects aspects: ProgramAspects, origin: ProgramOrigin) {
+        var aspectsToTrack = aspects
+
         func processCommon(_ program: Program) {
             let (newTypeCollectionRun, _) = updateTypeInformation(for: program)
 
             dispatchEvent(events.InterestingProgramFound, data: (program, origin, newTypeCollectionRun))
 
             // All interesting programs are added to the corpus for future mutations and splicing
-            corpus.add(program)
+            corpus.add(program, aspectsToTrack)
+        }
+
+        // If only adding deterministic samples, execute each interesting one a second time and check for shared aspects
+        if deterministicCorpus {
+            evaluator.resetAspects(aspects)
+            let execution = execute(program)
+            guard execution.outcome == .succeeded else { return }
+            guard let secondExecutionAspects = evaluator.evaluate(execution) else { return }
```

Comment:
I'm wondering if it would be nicer to have one method `evaluator.evaluateAndIntersect(execution, with: previousAspects)` which does all of
- Reset the aspects of the previous execution
- Evaluate the current execution
- Intersect the two aspects
- Reset the edges that were only triggered in one of the two executions
- Returns the intersection of the aspects

I think that should be possible, and would remove a lot of these slightly weird methods that otherwise need to be exposed by the Evaluator. What do you think?

Reply:
I think that makes sense.

As a small modification, I think it should also return whether or not any edges in the aspect provided had to be reset, which will make the logic for doing N rounds of execution more straightforward to ensure determinism.